sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileDestinationCheckpointData.cs

```diff
@@ -135,7 +135,7 @@ internal class ShareFileDestinationCheckpointData : StorageResourceCheckpointDat
             PreserveFileAttributes = fileAttributes?.Preserve ?? true;
 
             FilePermissionKey = filePermissionKey;
-            _filePermissionKeyBytes = !string.IsNullOrEmpty(FilePermissionKey) ? Encoding.UTF8.GetBytes(filePermissionKey) : Array.Empty<byte>();
+            _filePermissionKeyBytes = !string.IsNullOrEmpty(filePermissionKey) ? Encoding.UTF8.GetBytes(filePermissionKey) : Array.Empty<byte>();
```

Comment:
Because in the case that customers pause and then go to resume. Enumeration would have finished, and we would have already set the Destination Permission Key on the Destination Share. We need to save the permission key that's set on the destination share for this respective file, when we go to transfer, we already have the permission key.

Reply:
Remove Destination File Permission Key in checkpointer

Reply:
done